client/my-sites/themes/controller.jsx

```diff
@@ -70,10 +70,13 @@ export function loggedIn( context, next ) {
 		window.scrollTo( 0, 0 );
 	}
 
-	const Component = context.params.site_id ? SingleSiteComponent : MultiSiteComponent;
-	context.primary = <Component { ...getProps( context ) } />;
-
-	next();
+	if ( context.params.site_id ) {
+		context.primary = <SingleSiteComponent { ...getProps( context ) } />;
+		next();
+	} else {
+		// Force logged in user w/o a site to choose a site
+		return sites( context, next );
+	}
```

Comment:
The trouble with importing `sites` could be solved by moving the `loggedIn` handler to a separate module that is never imported on server. I tried to do that in #52813.

Rendering the site selector can be done, arguably more elegantly, with two distinct route handlers:
```js
page( '/themes', siteSelection, sites, render );
page( '/themes/:site', siteSelection, loggedIn, navigation, render );
```

This approach keeps each handler focused on exactly one task, and prevents rendering sidebar (`navigation`) in the site selection UI, which is currently a bug in this PR:

<img width="832" alt="Screenshot 2021-05-13 at 11 39 45" src="https://user-images.githubusercontent.com/664258/118108272-1584c100-b3e0-11eb-8cbd-25a4d536d45a.png">